src/shared/Microsoft.Git.CredentialManager.Tests/Interop/LibGit2Tests.cs

```diff
@@ -0,0 +1,201 @@
+            using (var config = git.GetConfiguration(repoPath))
+            {
+                string value = config.GetString("user.name");
+                Assert.NotNull(value);
+                Assert.Equal("john.doe", value);
+            }
+        }
+
+        [Fact]
+        public void LibGit2Configuration_GetRepositoryPath_ReturnsRepositoryPath()
+        {
+            string repoBasePath = CreateRepository();
+            string expectedRepoGitPath = Path.Combine(repoBasePath, ".git");
+            string fileL0Path = Path.Combine(repoBasePath, "file.txt");
+            string directoryL0Path = Path.Combine(repoBasePath, "directory");
+            string fileL1Path = Path.Combine(directoryL0Path, "inner-file.txt");
+            string directoryL1Path = Path.Combine(directoryL0Path, "sub-directory");
+
+            var git = new CredentialManager.Interop.LibGit2();
+
+            // Create files and directories
+            Directory.CreateDirectory(directoryL0Path);
+            Directory.CreateDirectory(directoryL1Path);
+            File.WriteAllText(fileL0Path, string.Empty);
+            File.WriteAllText(fileL1Path, string.Empty);
+
+            // Check from L0 file
+            string fileL0RepoPath = git.GetRepositoryPath(fileL0Path);
+            AssertPathsEquivalent(expectedRepoGitPath, fileL0RepoPath);
+
+            // Check from L0 directory
+            string dirL0RepoPath = git.GetRepositoryPath(directoryL0Path);
+            AssertPathsEquivalent(expectedRepoGitPath, dirL0RepoPath);
+
+            // Check from L1 file
+            string fileL1RepoPath = git.GetRepositoryPath(fileL1Path);
+            AssertPathsEquivalent(expectedRepoGitPath, fileL1RepoPath);
+
+            // Check from L1 directory
+            string dirL1RepoPath = git.GetRepositoryPath(directoryL1Path);
+            AssertPathsEquivalent(expectedRepoGitPath, dirL1RepoPath);
+        }
+
+        [Fact]
+        public void LibGit2Configuration_GetString_DoesNotExists_ReturnsNull()
+        {
+            string repoPath = CreateRepository();
+
+            var git = new CredentialManager.Interop.LibGit2();
+            using (var config = git.GetConfiguration(repoPath))
+            {
+                string randomKey = $"{Guid.NewGuid():N}.{Guid.NewGuid():N}";
+                string value = config.GetString(randomKey);
+                Assert.Null(value);
+            }
+        }
+
+        #region Test helpers
+
+        private static void AssertPathsEquivalent(string expected, string actual)
+        {
+            string realExpected = RealPath(expected);
+            string realActual = RealPath(actual);
+
+            Assert.Equal(realExpected, realActual);
+        }
+
+        private static string RealPath(string path)
+        {
+            if (PlatformUtils.IsPosix())
+            {
+                byte[] pathBytes = Encoding.UTF8.GetBytes(path);
+                byte[] resolvedBytes = new byte[64 * 1024];
```

Comment:
Is there a `PATH_MAX` we can use here for a maximum value (from a quick look at `realpath` documentation, it looks like it might be platform specific?)

Reply:
There is, and yes it's platform specific.

This method exists only in the specific LibGit2 tests and only used in here. In practice it's only going to be `<tmp-base>/repo-12345678/<1-10 long>` in total length.

If someone decides to move the temporary location to more than 64k characters deep, then they can fix the test. 😄

Reply:
I decided to use the returned pointer from `realpath` rather than the other 'output' mechanism via user allocated memory.

`realpath` returns a null pointer on error, or the resolved `char *` path. On success it also writes the resolved path to the user supplied `char *resolved_path`.